site/docs/tutorials/snake/drawing-the-snake.md

```diff
@@ -217,6 +217,66 @@ You should see some green blocks at the top.
 
 </Page>
 
+<Page value="wat">
+
+To draw the snake, you can loop through each point using the `loop` instruction. To make it a little easier, it's a good idea to use the `Rect` function of WASM-4:
+
+```wasm
+// Rect draws a rectangle. It uses color 1 to fill and color 2 for the outline
+(import "env" "rect" (func $rect (param i32 i32 i32 i32)))
+```
+
+With that out the way, let's see what a first draft could look like. In the WebAssembly text format, it's often simpler to use an address directly rather than an array index. However, since we know that the array starts at 0x19ac, we can use that as a direct index in the load instructions. So we'll actually use a byte offset to access each point of the snake.
+
+```wasm
+(func $snake-draw
+  (local $offset i32)      ;; offset is initialized to 0
+  (local $offset-end i32)
+
+  ;; offset-end = body_length * 8
+  (local.set $offset-end
+    (i32.mul
+      (i32.load (i32.const 0x19a8))  ;; body_length
+      (i32.const 8)))
+
+  ;; loop over all points in the body
+  (loop $loop
+    ;; rect(part.x * 8, part.y * 8, 8, 8)
+    ;; 
+    ;; Note that the array starts at 0x19a4, so the first x-coordinate starts
```

Comment:
Doesn't the array start at 0x19ac, not 0x19a4? If I understand correctly, from creating-the-snake.md, the offset into the snake.body array is 0x19ac.

Reply:
Yep, you're right. Thanks!